src/Symfony/Contracts/EventDispatcher/EventListenerInterface.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Contracts\EventDispatcher;
+
+/**
+ * Marker interface for event listeners.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+interface EventListenerInterface
+{
+    /*
+     * Handle an event.
+     *
+     * This method is required and the `$event` argument have to be Type-Hinted with the listened event's class in order
+     * to be auto-configured.
+     *
+     * @param object $event     The event The event object passed to the dispatcher
+     * @param string $eventName The name of the event dispatched
+     * @param EventDispatcherInterface $dispatcher The dispatcher used.
+     *
+     * @return void
+     *
+     * // minimal requirement
+     * public function __invoke(Event $event): void;
+     *
+     * // full signature
+     * public function __invoke(Event $event, string $eventName, EventDispatcherInterface $dispatcher): void;
```

Comment:
I would suggest to move this comment at the class level, and use two `@method` annotations:
`@method void __invoke(object $event, ...)`

Reply:
moved. Kept a single `@method` as user are free to remove unused parameters.